### PR TITLE
git-submodule-bump: Add helper for submodule bumps

### PR DESCRIPTION
**Pull request by zeenix**
Add a shell script that makes it easy to bump submodules. The script
create (most of) the commit message for you. You still need to provide the
short log but script takes care of prefix for you. Here is an example run:

---------
$ ./git-submodule-bump "Bump to latest git master"
[wip/subm-update-commit bfff560] meta-ivi: Bump to latest git master
 1 file changed, 1 insertion(+), 1 deletion(-)
---------

Signed-off-by: Zeeshan Ali <zeeshan.ali@pelagicore.com>

### CODE REVIEW

File: git-submodule-bump [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+MSG=$(echo "${SUBMODULE}: ${SHORTLOG}")
+MSG+=$'\n\n'
+MSG+=$(git -C ${SUBMODULE} log --format="%h %s" ${OLDCOMMIT}..${NEWCOMMIT})
+
+NAME=$(git config user.name)
+EMAIL=$(git config user.email)
+
+MSG+=$'\n\n'
+MSG+=$(echo "Signed-off-by: ${NAME} <${EMAIL}>")
```
Comment (chbae) [inline git-submodule-bump]: It's better to use "-s" option in git instead of upper 5 lines.

Comment (zeenix) [inline git-submodule-bump]: Good catch. I had forgotten about that option and have been manually signing. :) It's fixed now.

File: git-submodule-bump [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+SHORTLOG=${1}
+if test -z ${SHORTLOG}; then
+    echo -e "Usage:\n"
+    echo "${0} COMMIT_SUMMARY"
+    exit -1
+fi
+
+CHANGED_MOD=$(git submodule status | grep '^+' | tr -d + | head -n 1)
+if test -z ${SHORTLOG}; then
```
Comment (chbae) [inline git-submodule-bump]: I thought that ${SHORTLOG }should be changed to ${CHANGED_MOD}. right?

Comment (zeenix) [inline git-submodule-bump]: good catch, fixed!

File: git-submodule-bump [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+SHORTLOG=${1}
+if test -z ${SHORTLOG}; then
+    echo -e "Usage:\n"
+    echo "${0} COMMIT_SUMMARY"
+    exit -1
+fi
+
+CHANGED_MOD=$(git submodule status | grep '^+' | tr -d + | head -n 1)
+if test -z ${CHANGED_MOD}; then
```
Comment (chbae) [inline git-submodule-bump]: if test -z ${CHANGED_MOD}; then -> if test -z "${CHANGED_MOD}"; then

Comment (zeenix) [inline git-submodule-bump]: why do we need to quote the variable? We're not doing it in other places either.

Comment (chbae) [inline git-submodule-bump]: Please see http://unix.stackexchange.com/questions/86800/quote-a-variable-in-shell-script-if-statement and I got the variable error depended on system.

**Pull request merged**